File: backend/ShiftMate.API/Models/LeaveRequest.cs
namespace ShiftMate.API.Models;

public class LeaveRequest
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public string? Reason { get; set; }

    public string Status { get; set; } = "Pending";

    public DateTime CreatedAt { get; set; }
    
    public Employee Employee { get; set; } = null!;
}

File: frontend/src/types/leaveRequest.ts
export interface LeaveRequest {
    id: number
    employeeId: number
    employeeName: string
    startDate: string
    endDate: string
    reason: string | null
    status: string
    createdAt: string
}

export interface CreateLeaveRequest {
    employeeId: number
    startDate: string
    endDate: string
    reason: string
}